Building torch 2.9 nightly wheels fails on missing CMake deps

https://github.com/ROCm/TheRock/actions/runs/19029964424/job/54341342234
```
/__w/TheRock/TheRock/external-builds/pytorch/triton/setup.py:357: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.
  file.extractall(path=tmp_path)
/__w/TheRock/TheRock/external-builds/pytorch/triton/setup.py:314: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.
  file.extractall(path=package_root_dir)
creating symlink: /github/home/.triton/llvm/llvm-almalinux-x64 -> /github/home/.triton/llvm/llvm-7d5de303-almalinux-x64
CMake Deprecation Warning at CMakeLists.txt:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- The CXX compiler identification is GNU 13.3.1
-- The C compiler identification is GNU 13.3.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/gcc-toolset-13/root/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/rh/gcc-toolset-13/root/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11")
-- Could NOT find zstd (missing: zstd_LIBRARY zstd_INCLUDE_DIR) 
CMake Error at CMakeLists.txt:105 (find_package):
  Found package configuration file:

    /github/home/.triton/llvm/llvm-7d5de303-almalinux-x64/lib/cmake/mlir/MLIRConfig.cmake

  but it set MLIR_FOUND to FALSE so package "MLIR" is considered to be NOT
  FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing:
  LLVMNVPTXCodeGen LLVMNVPTXDesc LLVMNVPTXInfo



-- Configuring incomplete, errors occurred!
```

## Reproducer
### Fetch projects
```bash
git clone https://github.com/ROCm/TheRock.git
cd TheRock
python3 ./external-builds/pytorch/pytorch_torch_repo.py checkout --gitrepo-origin https://github.com/ROCm/pytorch.git --repo-hashtag release/2.9 --patchset rocm_2.9
python3 ./external-builds/pytorch/pytorch_triton_repo.py checkout
```

### Build trition with TheRock `7.10.0a20251028` (passing)
1. Install ROCm
```bash
python3 -m venv venv-20251028
source venv-20251028/bin/activate
pip install rocm[libraries,devel]==7.10.0a20251028 -i https://rocm.nightlies.amd.com/v2/gfx94X-dcgpu/
```

2. Set environment variables and build trition wheel
```bash
export CMAKE_PREFIX_PATH=`rocm-sdk path --cmake`
export ROCM_HOME=`rocm-sdk path --root`
export ROCM_PATH=`rocm-sdk path --root`
export HIP_DEVICE_LIB_PATH="`rocm-sdk path --root`/ib/llvm/amdgcn/bitcode"
export PYTORCH_ROCM_ARCH=gfx942
export TRITON_WHEEL_NAME=pytorch-triton-rocm

pushd external-builds/pytorch/triton/
pip install -e .
```

3. Check CMake cache
(Adjust path if not building for Python 3.12)
```bash
grep LLVM_ build/cmake.linux-x86_64-cpython-3.12/CMakeCache.txt
```
shows
```
LLVM_DIR:PATH=/.triton/llvm/llvm-7d5de303-ubuntu-x64/lib/cmake/llvm
LLVM_ENABLE_WERROR:UNINITIALIZED=ON
LLVM_EXTERNAL_LIT:STRING=
LLVM_INCLUDE_DIRS:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64/include
LLVM_LIBRARY_DIR:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64/lib
LLVM_SYSPATH:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64
...
```
### Cleanup
```bash
rm -rf build
popd
deactivate
```
### Build trition with TheRock `7.10.0a202511042` (failing)
1. Install ROCm
```bash
python3 -m venv venv-20251104
source venv-20251104/bin/activate
pip install rocm[libraries,devel]==7.10.0a20251104 -i https://rocm.nightlies.amd.com/v2/gfx94X-dcgpu/
```

2. from above fails with
```
      CMake Error at CMakeLists.txt:105 (find_package):
        Found package configuration file:
      
          /.triton/llvm/llvm-7d5de303-ubuntu-x64/lib/cmake/mlir/MLIRConfig.cmake
      
        but it set MLIR_FOUND to FALSE so package "MLIR" is considered to be NOT
        FOUND.  Reason given by package:
      
        The following imported targets are referenced, but are missing:
        LLVMNVPTXCodeGen LLVMNVPTXDesc LLVMNVPTXInfo
```

3. from above shows
```
LLVM_DIR:PATH=/tmp/TheRock/venv-20251104/lib/python3.12/site-packages/_rocm_sdk_devel/lib/cmake/llvm
LLVM_ENABLE_WERROR:UNINITIALIZED=ON
LLVM_INCLUDE_DIRS:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64/include
LLVM_LIBRARY_DIR:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64/lib
LLVM_SYSPATH:UNINITIALIZED=/.triton/llvm/llvm-7d5de303-ubuntu-x64
```
## Summary
With `7.10.0a202511042` the `LLVM_DIR` variable points to TheRock's LLVM and _not_ the LLVM packages provided with trition. The former does not provide the `LLVMNVPTXCodeGen`, `LLVMNVPTXDesc`, and `LLVMNVPTXInfo` target and thus the build fails.

--- a/external-builds/pytorch/build_prod_wheels.py
+++ b/external-builds/pytorch/build_prod_wheels.py
@@ -397,9 +397,9 @@ def do_build(args: argparse.Namespace):
         # Here, `CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH` is set
         # and passed via `TRITON_APPEND_CMAKE_ARGS` to avoid this.
         # See also https://github.com/ROCm/TheRock/issues/1999.
-        env["TRITON_APPEND_CMAKE_ARGS"] = (
-            "-DCMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH=FALSE"
-        )
+        env[
+            "TRITON_APPEND_CMAKE_ARGS"
+        ] = "-DCMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH=FALSE"
 
     if is_windows:
         llvm_dir = rocm_dir / "lib" / "llvm" / "bin"

